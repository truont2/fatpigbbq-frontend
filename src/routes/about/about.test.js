import { render, screen } from "@testing-library/react";
import About from "./about";
import TestRenderer from 'react-test-renderer';

describe("About Page", () => {
  it("renders correctly", () => {
    const component = TestRenderer.create(<About />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should display the about page header", () => {
    render(<About />);
    const pageTitle = screen.getByText("About Us").textContent;
    expect(pageTitle).toBe("About Us");
  });


  it("should display page text", () => {
    // Render the component
    render(<About />);
    const element = screen.getByTestId("custom-element").textContent;
    expect(element).toBe(
      "Fat Pig BBQ was established in 2018 by Bao Truong. Bao has always enjoyed cooking barbeque for his family & friends at his home. Bao later realized that he wanted share his passion to provide great barbeque to everyone in the Greater Seattle Area. Shortly after, Bao left his Civil Engineering job and opened the doors to Fat Pig BBQ in Edmonds."
    );
  });
});
