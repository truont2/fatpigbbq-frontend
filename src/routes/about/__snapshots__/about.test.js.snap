// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page renders correctly 1`] = `
<div>
   
  <div
    className="relative w-screen bg-black overflow-hidden"
  >
    <div
      className="bg-fixed bg-repeat-no bg-cover bg-left-bottom w-screen md:h-[250px] h-[100px] object-cover object-center opacity-70"
      style={
        Object {
          "backgroundImage": "url(meal.jpeg)",
        }
      }
    />
    <div
      className="absolute text-5xl text-white top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2"
    >
      <h1
        className="md:visible invisible md:text-4xl text-4xl font-medium title-font text-center font-CooperBlack"
      >
        About Us
      </h1>
    </div>
  </div>
  <div
    className="max-w-[1400px] m-auto pt-6 px-4 flex justify-center flex-col items-center text-gray-500"
  >
    <div
      className=" text-center lg:col-span-2 col-span-2 my-10"
    >
      <h3
        className="text-5xl font-bold"
      >
        The Story of Fat Pig 
      </h3>
      <p
        className="pt-10 text-2xl"
        data-testid="custom-element"
      >
        Fat Pig BBQ was established in 2018 by Bao Truong. Bao has always enjoyed cooking barbeque for his family & friends at his home. Bao later realized that he wanted share his passion to provide great barbeque to everyone in the Greater Seattle Area. Shortly after, Bao left his Civil Engineering job and opened the doors to Fat Pig BBQ in Edmonds.
      </p>
    </div>
    <div
      className="h-[400px]"
    >
      <img
        alt="/"
        className="object-cover w-full h-full bg-fixed bg-no-repeat bg-auto  bg-center"
        src="o.jpg"
      />
    </div>
  </div>
  <div
    className=" m-auto max-w-[1400px] py-16 px-4 grid grid-cols-2 lg:grid-cols-5 gap-2"
  >
    <img
      alt={0}
      className="rounded-md"
      src="DSC_7011.jpg"
    />
    <img
      alt={1}
      className="rounded-md"
      src="DSC_7015.jpg"
    />
    <img
      alt={2}
      className="rounded-md"
      src="DSC_7018.jpg"
    />
    <img
      alt={3}
      className="rounded-md"
      src="DSC_7022.jpg"
    />
    <img
      alt={4}
      className="rounded-md"
      src="porkMeal.jpg"
    />
  </div>
</div>
`;
